### DISCUSSION TITLE
feature request : search for abbreviations

### DISCUSSION THREAD

**Discussion by matthiasdebernardini** [PatrickF1/fzf.fish#90] [Archive]
A quick hotkey for looking at your abbrs and functions.

this here works for me pretty well, would this be hard to add or out of scope for the project?

I currently have the command below abbreviated to `gabb` for get abbr. 

`bat .config/fish/conf.d/abbreviations.fish | rg abbr | fzf`

**Comment by PatrickF1** [comment]
Hi Mattias, I like your creativity! I am unfamiliar with fish abbreviations. Are they always stored in `.config/fish/conf.d/abbreviations.fish`? I think using fzf to quickly find abbreviations is a great idea but only if we can make it work consistently and reliably across all platforms and set ups.

**Comment by matthiasdebernardini** [comment]
Hi Patrick,

Thanks!

Fish abbreviations are essentially aliases done the fish way.

An alias can be 
`alias gc="git checkout"`
whereas an abbr can be
`abbr gc "git checkout"`

However, abbr's substitute "in place" so your history is a bit cleaner and IMO is a bit more clear whats going on, so it's easier to read.

They are not always stored in this file name, so I should probably change that. 

Maybe I can grep for every instance of abbr in every file that ends with `.fish`

This could also be used to look at aliases, since I usually forget them when I am trying to learn them.

with regards to the interoperability, do you have fd installed on your Mac?

I am using fd-find from rust (because its faster and has simpler syntax)

however I don't know how this would work on windows

**Comment by PatrickF1** [comment]
Hi Matt,

> They are not always stored in this file name, so I should probably change that.
Maybe I can grep for every instance of abbr in every file that ends with .fish

That's a lot better than assuming the user always stores it in that exact file. However, I still think that's a non-starter due to
- the near impossibility of finding every single .fish file on the system
- the performance impact

> This could also be used to look at aliases, since I usually forget them when I am trying to learn them.

Sounds like a slightly rare but interesting case.

> do you have fd installed on your Mac?

Yes, if you didn't know, fzf.fish requires `fd` for the find file functionality!

> however I don't know how this would work on windows

Why wouldn't it work on Windows?

Overall I am a bit hesitant about adding this feature due to how rarely people will use it and how difficult it is to implement. But let me think about it more and get back to you! I really appreciate your enthusiasm and am sorry that I am telling you this now after you have already opened a PR.

**Comment by matthiasdebernardini** [comment]
Hi, no problem I enjoyed working on this for myself. 

> the near impossibility of finding every single .fish file on the system

They should all be in one folder, the `.config/fish/` folder. Even the most insanse dotfiles only have like 10 or 20 files. If you use the rust version of fd (which handles concurrency correctly) its decent in terms of speed.

> Why wouldn't it work on Windows?

I don't know how windows handles these things but usually nothing works on windows without extra work and edge cases. 

I totally understand, I do have an abbreviation for listing my abbreviations and I think thats just the best way currently so I will continue to use it. I will close it.

**Comment by PatrickF1** [comment]
> Hi, no problem I enjoyed working on this for myself.

Hey Matt, thanks so much for understanding. I hate to see you lose all the work you excitedly pour down but at the same time it's also my job as the plugin author to defend against feature cruft. I hope you'll gain a lot from your experience and become a plugin author yourself!

> I totally understand, I do have an abbreviation for listing my abbreviations and I think thats just the best way currently so I will continue to use it. I will close it.

Btw, I was studying the abbreviations command. Why don't you use `abbr --list` instead of cating your abbreviations file?

**Comment by matthiasdebernardini** [comment]
Of course. You being able to say No to unnecessary features is what makes this project so good. In the past couple of days I have used this maybe 100 times. It is so useful and I'm happy that I found it. 

I dont use abbr --list becuase it just shows the abbr and not what its abbreviating. Also I didn't know about it xD

**Comment by luong-komorebi** [comment]
+1 for this answer. Currently, it takes a lot of effort to search for an abbreviation and often requires reading the source code of a plugin
